### PR TITLE
fix: Address `.numpy()` issue on fake tensors

### PR DESCRIPTION
**Pull request by gs-olive**
# Description

- Add `fake_tensor_unsupported` decorator to helper backend
- Refactor `conversion` implementation to use compilation settings object as well, to reduce code duplication and encourage reuse
- Improve debugger messages by pre-formatting support string

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ x ] I have added tests to verify my fix or my feature
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/backend/backends.py [hunk 1]
```diff
@@ -52,6 +52,7 @@ def aot_torch_tensorrt_aten_backend(
     )
 
 
+@fake_tensor_unsupported
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/backend/backends.py:55]: This is the ultimate backend everything goes through right? so does that mean we cant work on fake tensors? is this different than symbolic shapes?

Comment (narendasan) [reply py/torch_tensorrt/dynamo/backend/backends.py:55]: Also could we just turn a "fake_tensor" into an "ITensor" immediately. sounds like they are similar

Comment (gs-olive) [reply py/torch_tensorrt/dynamo/backend/backends.py:55]: Yes this is the backend everything goes through, and my understanding is that `FakeTensors` are for use at compile time, and are distinct from `SymInt`s which are the symbolic shape representations.

The challenge with fake tensors right now is that _any_ tensors instantiated during the call are fake, which means that the constant tensors which we need to provide to TensorRT, as in the code snippet below, are "fake" and thus contain no value to be parsed. I think a long-term solution could be to support Fake Tensors fully, but this solution temporarily resolves TRT/Torch compatibility issues.

https://github.com/pytorch/TensorRT/blob/c7e79b2f3f48a1bd47266077f414117e42ba2ff5/py/torch_tensorrt/fx/converters/converter_utils.py#L247-L255

Comment (gs-olive) [reply py/torch_tensorrt/dynamo/backend/backends.py:55]: Instantiated #1951 with a feature proposal and additional discussion

**Pull request merged**